ports/esp8266/esppwm.c

```diff
@@ -188,14 +195,16 @@ pwm_start(void) {
         // start
         gpio_output_set(local_single[0].gpio_set, local_single[0].gpio_clear, pwm_gpio, 0);
 
+        // do the first toggle because timer has to have a valid set to do it's job
+        pwm_toggle ^= 0x01;
+
         pwm_timer_down = 0;
         RTC_REG_WRITE(FRC1_LOAD_ADDRESS, local_single[0].h_time);
-    }
-
-    if (pwm_toggle == 1) {
-        pwm_toggle = 0;
     } else {
-        pwm_toggle = 1;
+        // request for toggle
+        irq_state = mp_hal_quiet_timing_enter();
+        do_toggle = 1;
+        mp_hal_quiet_timing_exit(irq_state);
```

Comment:
As above, I don't think the disabling of IRQs is needed here.

Reply:
Same, OK.